### ISSUE TITLE
in_app_purchase package unable to load products from Play Store

### ISSUE THREAD

**Issue by Yudeg1128** [flutter/flutter#134812]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Followed steps here exactly: https://github.com/flutter/packages/blob/main/packages/in_app_purchase/in_app_purchase/example/README.md
2. In-app product is active in the Pla console: 
![image](https://github.com/flutter/flutter/assets/67855177/bbb69a10-1261-4432-9d56-b771a373b5b2)
3. The .aab has the correct permissions: 
![image](https://github.com/flutter/flutter/assets/67855177/0585686f-d8e2-4cda-90d2-9e7e9a4631f3)
4. License tester is setup.
5. License tester follows the provded link, downloads the test bundle and the app says 'consumable' not found.

### Actual results

I am trying to implement the in_app_purchase package example app, followed all the steps in the example readme file. The app says the store is available, but cannot find the product by its ID. I did follow all of the steps specified in the example. Here: https://github.com/flutter/packages/blob/main/packages/in_app_purchase/in_app_purchase/example/README.md, even here: https://codelabs.developers.google.com/codelabs/flutter-in-app-purchases#4.

Cannot figure out what is wrong, tried for weeks to fix myself but nothing works.

### Logs

<details>
<summary>Logs</summary>

```console

<!-- Paste your logs here -->

```

</details>

### Flutter Doctor output

<details>
<summary>Doctor output</summary>

```console

[✓] Flutter (Channel stable, 3.13.2, on Pop!_OS 22.04 LTS 6.4.6-76060406-generic, locale en_US.UTF-8)
    • Flutter version 3.13.2 on channel stable at /usr/local/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ff5b5b5fa6 (3 weeks ago), 2023-08-24 08:12:28 -0500
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /home/me/Android/Sdk
    • Platform android-33-ext5, build-tools 33.0.2
    • Java binary at: /usr/local/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 14.0.0-1ubuntu1.1
    • cmake version 3.22.1
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2022.2)
    • Android Studio at /usr/local/android-studio
    • Flutter plugin version 73.0.2
    • Dart plugin version 222.4560
    • android-studio-dir = /usr/local/android-studio
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Pop!_OS 22.04 LTS 6.4.6-76060406-generic

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

```

</details>

**Comment by danagbemava-nc** [comment]
Hi @Yudeg1128, is the licensed tester signed into playstore on the device that you're testing on? Do you get any errors and can you share the errors?

**Comment by Yudeg1128** [comment]
> Hi @Yudeg1128, is the licensed tester signed into playstore on the device that you're testing on? Do you get any errors and can you share the errors?

Hi, thanks for response! Yes I have only one License tester, myself, and logged on to Google Play with the Google account registered for the license tester(on my physical device). I do not get any errors. Here is the log after flutter run:

Launching lib/main.dart on M2102J20SG in debug mode...
Running Gradle task 'assembleDebug'...
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Debug service listening on ws://127.0.0.1:37935/aqxHsCDdxY0=/ws
Syncing files to device M2102J20SG...
D/DecorView[]( 7234): onWindowFocusChanged hasWindowFocus true

I also attached the Sample app screenshot.

![signal-2023-09-18-171917_002](https://github.com/flutter/flutter/assets/67855177/a81d7acc-8688-49ea-91ea-543d5ac64dfc)

Here is a current screenshot of the product active in the Play console:

![image](https://github.com/flutter/flutter/assets/67855177/df93992e-7e5d-400e-b447-6a0ab79272b8)


Right now the Sample app is published in the Internal testing track in the Play console. From what I understand, a License tester can test billing even in internal testing with Closed Alpha and review of the app?

**Comment by danagbemava-nc** [comment]
Hi @Yudeg1128, can you try removing the ids of the products you do not have yet to see if that helps?

**Comment by Yudeg1128** [comment]
Yes, those products already exist.

**Comment by huycozy** [comment]
@Yudeg1128 Can you try creating a new ID on the console (`test_consumable` for eg), then update it to `_kConsumableId` value on the sample code and retry?

In case you logged many Google accounts to the device, in order to make sure it's the correct tester on your device, you can try removing all other Google accounts, keep only one tester account, and check this again.

**Comment by Yudeg1128** [comment]
> @Yudeg1128 Can you try creating a new ID on the console (`test_consumable` for eg), then update it to `_kConsumableId` value on the sample code and retry?
> 
> In case you logged many Google accounts to the device, in order to make sure it's the correct tester on your device, you can try removing all other Google accounts, keep only one tester account, and check this again.

Thank you for response. The product ID in the Play console is permanent and cannot be changed or updated. However, I did an update on the product name like so:

![image](https://github.com/flutter/flutter/assets/67855177/461a8a02-248c-46ab-a737-05116aa0e177)

Also, removed all google accounts, logged in as license tester and still got same result:

![image](https://github.com/flutter/flutter/assets/67855177/233280db-e881-4265-a5ac-f31ce325acdc)

Does the example app still work? Am I the only one unable to get it to work?

**Comment by danagbemava-nc** [comment]
Hi @Yudeg1128, what region(s) are the consumables set for and what region are you accessing the app from? I just updated my old iap example to use the latest source code and I can access my consumables and subscriptions just fine.

**Comment by Yudeg1128** [comment]
> Hi @Yudeg1128, what region(s) are the consumables set for and what region are you accessing the app from? I just updated my old iap example to use the latest source code and I can access my consumables and subscriptions just fine.

Thank you for your response, yes I think I did everything according to official guides and it should work now, very perplexed.

These are the locations my app bundle is released for:
![image](https://github.com/flutter/flutter/assets/67855177/2952c561-849c-4662-babc-62224f11265b)

I am accessing the app from Mongolia, which is included in the app bundle locations as mn.

Am I missing something with the app IDs? I made sure all the IDs everywhere match each other like this:

![image](https://github.com/flutter/flutter/assets/67855177/c3884ee6-f9d4-4c68-8800-5fadb2cdf7b4)

In my build.gradle:

![image](https://github.com/flutter/flutter/assets/67855177/0613f346-1079-4e61-98bc-90b66ed5bd88)

And in my AndroidManifest.xml, I have not specified package name, I assume that makes no difference?

**Comment by danagbemava-nc** [comment]
Hi @Yudeg1128, is the apk you're using signed? Also, for subscriptions, you can specify regions in which it should be active. Is your current region listed in the subscriptions? Or are you only testing consumables?

**Comment by Yudeg1128** [comment]
> Hi @Yudeg1128, is the apk you're using signed? Also, for subscriptions, you can specify regions in which it should be active. Is your current region listed in the subscriptions? Or are you only testing consumables?

Hi!

1. Yes, I did sign the app using the guides here: https://docs.flutter.dev/deployment/android

However, when uploading the app bundle, I chose the releases to be signed by Google as it was the recommended option:

![image](https://github.com/flutter/flutter/assets/67855177/f91f7314-2d75-4df4-b641-33179073385b)

From what I understand, this shouldn't create any conflict, since these are 2 different procedures?

2. My app only needs a couple of consumable IAPs, so I never tested any subscriptions. Are subscriptions required to get the package to work?

Thank you.
